### ISSUE TITLE
tweet length incorrect. Incorrectly assumes url

### ISSUE THREAD

**Issue by Riuujin** [linvi/tweetinvi#594]
When testing the length of a tweet with the following text: "First sentence.Second sentence."
It returns incorrectly that the length is '39',it should be '31' according to twitters own library (https://github.com/twitter/twitter-text).

This happens because "sentence.Second" is seen as a URL.

**Comment by linvi** [comment]
Thank you for reporting this bug, I will look into it.

**Comment by linvi** [comment]
I would like to say that the algorithm seems to have changed since I last look into it as `google.fr` is no longer matched as a url.

**Comment by Riuujin** [comment]
That is weird.
Also not there read me warning: "⚠️ Note that a new version of twitter-text will be released soon"

**Comment by linvi** [comment]
Yes this new version will include various new rules and new support for 280 characters.

**Comment by linvi** [comment]
Opened issue to get additional information : https://github.com/twitter/twitter-text/issues/200

**Comment by ayaz-ashrapov** [comment]
Hi @linvi, sorry to bother that I saw that Twitter has updated their twitter-text implementation and fixed their tests for CTLDs. Did you check it already by any chance?

**Comment by linvi** [comment]
I have seen that they are supposed to have fixed it. But it does not mean that it has been rollout on live.
I will check this out as soon as I have time.

**Comment by linvi** [comment]
Hello,

I had a look at the changes performed for Twitter length calculation and unfortunately the algorithm used by Twitter has changed significantly resulting in Tweetinvi's calculation to no longer be accurate in various cases.

As a result I will have to investigate other solutions to calculate the length of a Tweet. As it turns out to be a complex matter I will not be able to do so for the next release which has already been delayed for too long.

For your project, I would advise using one of the official implementation for the moment, potentially the node.js as a micro service.

I will make the method as `obsolete` for the time being.

Sorry about this.

Regards,
Linvi

**Comment by linvi** [comment]
Use twitter-text implementation until new release.
